src/main/java/edu/ucsb/ucsbcslas/controllers/ActiveQuarterController.java

```diff
@@ -0,0 +1,98 @@
+package edu.ucsb.ucsbcslas.controllers;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import javax.validation.Valid;
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.interfaces.DecodedJWT;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import edu.ucsb.ucsbcslas.advice.AuthControllerAdvice;
+import edu.ucsb.ucsbcslas.entities.ActiveQuarter;
+import edu.ucsb.ucsbcslas.repositories.ActiveQuarterRepository;
+
+@RestController
+public class ActiveQuarterController {
+  private final Logger logger = LoggerFactory.getLogger(ActiveQuarterController.class);
+  @Autowired
+  private AuthControllerAdvice authControllerAdvice;
+  @Autowired
+  private ActiveQuarterRepository activeQuarterRepo;
+
+  private ObjectMapper mapper = new ObjectMapper();
+
+  private ResponseEntity<String> getUnauthorizedResponse(String roleRequired) throws JsonProcessingException {
+    Map<String, String> response = new HashMap<String, String>();
+    response.put("error", String.format("Unauthorized; only %s may access this resource.", roleRequired));
+    String body = mapper.writeValueAsString(response);
+    return new ResponseEntity<String>(body, HttpStatus.UNAUTHORIZED);
+  }
+
+  @PostMapping(value = "/api/admin/filter/{activeValue}", produces = "application/json")
+  public ResponseEntity<String> createFilter(@RequestHeader("Authorization") String authorization,
+      @PathVariable("activeValue") String activeValue) throws JsonProcessingException {
+    if (!authControllerAdvice.getIsAdmin(authorization))
+      return getUnauthorizedResponse("admin");
+    activeQuarterRepo.deleteAll();
+    
+    
+    
+    ActiveQuarter active = new ActiveQuarter();
+    active.setActiveQuarter(activeValue);
+    activeQuarterRepo.save(active);
+    String body = mapper.writeValueAsString(active);
+
+    return ResponseEntity.ok().body(body);
+  }
+
+  @DeleteMapping(value = "/api/admin/filter/nuke", produces = "application/json")
```

Comment:
I'd change this endpoint to just `/api/admin/filter`.

Reply:
I'll just get rid of this all together since a deleteall is used in the post mapping anyhow for upsert reasons.